lib/Runtime/Base/ScriptContext.h

```diff
@@ -2034,16 +2034,10 @@ namespace Js
     ArenaAllocator * allocator = nullptr;
 
 #define ACQUIRE_TEMP_GUEST_ALLOCATOR(allocator, scriptContext, name) \
-    TryFinally([&]() \
-    { \
-        tempGuest##allocator = scriptContext->GetTemporaryGuestAllocator(name); \
```

Comment:
This is reverting my previous fix as it is unnecessary. The root cause was that we were not unpinning the guest arena after parser was done with it.